Bundle/CoreBundle/Entity/View.php

```diff
@@ -554,22 +553,28 @@ public function removeWidgetMap(WidgetMap $widgetMap)
     }
 
     /**
-     * Get widgets ids as array.
+     * Get WidgetMaps for View and Templates.
      *
-     * @return array
+     * @return WidgetMap[]
      */
-    public function getWidgetsIds()
+    public function getWidgetMapsForViewAndTemplates($viewContext = null)
```

Comment:
I think viewContext is not a good name as I don't understand what it is... I expect it to be a new kind of object like `ViewContext` but here, we just talk about `View`. May `contextualView` (or even just `View`)  be a better name for you ?